### PR TITLE
Features & bug fixes

### PR DESCRIPTION
**Pull request by jgeramb**
I made a few more changes to the script:

- bug fixes
- distance text
- changed German to English text
- made fake profiles invisible

### CODE REVIEW

**Comment by tajnymag** [comment]
Looks great. Thank you!

I have a nitpicky question though. Why do you insist on using var for variable declaration?

**Comment by jgeramb** [comment]
Uhm I used const most of the time, maybe I forgot to use it sometimes 😅

File: tinder.user.js [hunk 1]
```diff
@@ -345,65 +363,57 @@ function updateUserInfos() {
...
-		photoSelectorContainer.setAttribute(
-			'class',
-			'photo-selectors CenterAlign D(f) Fxd(r) W(100%) Px(8px) Pos(a) Iso(i)'
-		);
+		photoSelectorContainer.setAttribute('class', 'photo-selectors CenterAlign D(f) Fxd(r) W(100%) Px(8px) Pos(a) Iso(i)');
 		photoSelectorContainer.style.top = '5px';
 		likeEl.appendChild(photoSelectorContainer);
 
-		for (let i = 0; i < user.photos.length; i++) {
+		for (var i = 0; i < user.photos.length; i++) {
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with let

File: tinder.user.js [hunk 2]
```diff
@@ -499,26 +511,19 @@ function updateUserFiltering() {
...
 						applyContainer.className = '';
-						applyContainer.setAttribute(
-							'style',
-							'align-items: center; display: flex; flex-shrink: 0; font-size: 20px; height: 50px; justify-content: center; width: 100%;'
-						);
+						applyContainer.setAttribute('style', 'align-items: center; display: flex; flex-shrink: 0; font-size: 20px; height: 50px; justify-content: center; width: 100%;');
 
-						const applyButtonEl = document.createElement('button');
-						applyButtonEl.innerText = 'Anwenden';
+						var applyButtonEl = document.createElement('button');
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 3]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+								const dialogMenuItemContents = document.querySelectorAll('div[role="dialog"] > .menuItem > .menuItem__contents > div:nth-of-type(2)');
 
 								// max distance
-								const maxDistanceElement = dialogMenuItemContents[0].querySelector('div[style]');
-								let maxDistance = Math.floor(
-									(maxDistanceElement.clientWidth / maxDistanceElement.parentElement.clientWidth) *
-										(161 - 2) +
-										2
-								);
+								var maxDistanceElement = dialogMenuItemContents[0].querySelector('div[style]');
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 4]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+
+								if (!maxDistanceElement) return;
+
+								var maxDistance = Math.floor((maxDistanceElement.clientWidth / (maxDistanceElement.parentElement?.clientWidth ?? 1)) * (161 - 2) + 2);
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with let

File: tinder.user.js [hunk 5]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
-										(100 - 18) +
-										18
-								);
-								let ageRangeEnd =
-									ageRangeStart +
-									Math.round(
-										(ageRangeElement.clientWidth / ageRangeElement.parentElement.clientWidth) *
-											(100 - 18)
-									);
+								var ageRangeElement = dialogMenuItemContents[1].querySelector('div[style]');
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 6]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+
+								if (!ageRangeElement) return;
+
+								var ageRangeStart = Math.round((parseFloat(getComputedStyle(ageRangeElement).left.replace('px', '')) / (ageRangeElement.parentElement?.clientWidth ?? 1)) * (100 - 18) + 18);
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 7]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+								var ageRangeEnd = ageRangeStart + Math.round((ageRangeElement.clientWidth / (ageRangeElement.parentElement?.clientWidth ?? 1)) * (100 - 18));
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with let

File: tinder.user.js [hunk 8]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
-
-								for (const minimumPhotosOption of dialogMenuItemContents[2].querySelectorAll(
-									'div[role="option"]'
-								)) {
-									if (
-										minimumPhotosOption
-											.getAttribute('class')
-											.includes('c-ds-border-passions-shared')
-									) {
+								var minimumPhotosAmount = 0;
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with let

File: tinder.user.js [hunk 9]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+								for (const minimumPhotosOption of photosOptions) {
+									if (minimumPhotosOption.getAttribute('class')?.includes('c-ds-border-passions-shared')) {
 										minimumPhotosAmount = parseInt(minimumPhotosOption.innerText);
 										break;
 									}
 								}
 
 								// interests
-								const interests = [];
+								var interests = [];
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 10]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
 
-								const dialogMenuSelects = document.querySelectorAll(
-									'div[role="dialog"] > .menuItem > .menuItem__contents .menuItem__select input'
-								);
+								/** @type {NodeListOf<HTMLInputElement>} */
+								const dialogMenuSelects = document.querySelectorAll('div[role="dialog"] > .menuItem > .menuItem__contents .menuItem__select input');
 
 								// verified
-								const verifiedRequired = dialogMenuSelects[0].checked;
+								var verifiedRequired = dialogMenuSelects[0].checked;
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 11]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
 
 								// has bio
-								const bioRequired = dialogMenuSelects[1].checked;
+								var bioRequired = dialogMenuSelects[1].checked;
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with const

File: tinder.user.js [hunk 12]
```diff
@@ -529,91 +534,79 @@ function updateUserFiltering() {
...
+									if (!userItem) continue;
+
 									const user = userItem.user;
-									const userInterests = Array.from(user.user_interests ?? []).map(
-										(interest) => interest.name
-									);
+									const userInterests = Array.from(user.user_interests ?? []).map((interest) => interest.name);
 
-									let matches = true;
+									var matches = true;
```
Comment (tajnymag) [inline tinder.user.js]: pls, replace with let

**Review by tajnymag — CHANGES_REQUESTED** [review]
I think I found all occurrences of var and haven't misjudged choice of const/let.

**Comment by jgeramb** [comment]
looks good to me

**Review by tajnymag — CHANGES_REQUESTED** [review]
It seems these are the last ones

**Review by tajnymag — APPROVED** [review]

**Comment by tajnymag** [comment]
LGTM, life's short, let's merge this 😎

**Pull request merged**